### PR TITLE
Unirule rest get by id

### PR DESCRIPTION
**Pull request by ahmadshadab**
Sample response:
https://gist.githubusercontent.com/ahmadshadab/0b53ddb0f8a934b13384587b15921767/raw/88a2895aa550fa2ac4ac550ad264255b1b5fde29/UniRule%2520Response

### CODE REVIEW

File: unirule-rest/src/main/java/org/uniprot/api/unirule/service/UniRuleSolrQueryConfig.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Bean
+    public SearchFieldConfig uniRuleSearchFieldConfig() {
+        return SearchFieldConfigFactory.getSearchFieldConfig(UniProtDataType.UNIRULE);
+    }
+
+    @Bean
+    public QueryProcessor uniRuleQueryProcessor(
+            WhitelistFieldConfig whiteListFieldConfig, SearchFieldConfig uniRuleSearchFieldConfig) {
+        Map<String, String> uniParcWhiteListFields =
```
Comment (LeonardoGonzales) [inline unirule-rest/src/main/java/org/uniprot/api/unirule/service/UniRuleSolrQueryConfig.java:41]: update variable name to uniRule

File: unirule-rest/src/test/java/org/uniprot/api/unirule/controller/UniRuleGetIdControllerIT.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        return "/unirule/";
+    }
+
+    static class UniRuleGetIdParameterResolver extends AbstractGetIdParameterResolver {
+
+        @Override
+        public GetIdParameter validIdParameter() {
+            return GetIdParameter.builder()
+                    .id(UNIRULE_ID)
+                    .resultMatcher(jsonPath("$.uniRuleId", is(UNIRULE_ID)))
```
Comment (LeonardoGonzales) [inline unirule-rest/src/test/java/org/uniprot/api/unirule/controller/UniRuleGetIdControllerIT.java:106]: Can you please check more fields here.

**Review by LeonardoGonzales — COMMENTED** [review]
Excellent. few comments only

File: unirule-rest/src/main/java/org/uniprot/api/unirule/controller/UniRuleController.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    @GetMapping(
+            value = "/{uniruleid}",
+            produces = {
+                TSV_MEDIA_TYPE_VALUE,
+                LIST_MEDIA_TYPE_VALUE,
+                APPLICATION_JSON_VALUE,
+                XLS_MEDIA_TYPE_VALUE
+            })
+    public ResponseEntity<MessageConverterContext<UniRuleEntry>> getByUniRuleId(
```
[the remark this answers is not in the document]
Comment (ahmadshadab) [reply unirule-rest/src/main/java/org/uniprot/api/unirule/controller/UniRuleController.java:61]: I will do when i generate and deploy the UniRule API on wwdev

**Pull request merged**